From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Thu, 26 Aug 2021 00:58 +0100
Subject: [PATCH] cloud-init: Adjust Alpine apk upgrade call

Add "--available" and "--update-cache" to Alpine "apk upgrade" call.

----

diff -aur a/cloudinit/distros/alpine.py b/cloudinit/distros/alpine.py
--- a/cloudinit/distros/alpine.py
+++ b/cloudinit/distros/alpine.py
@@ -128,6 +128,9 @@
         if command:
             cmd.append(command)
 
+        if command == 'upgrade':
+            cmd.extend(["--update-cache", "--available"])
+
         pkglist = util.expand_package_list('%s-%s', pkgs)
         cmd.extend(pkglist)
 
